src/libcore/num/mod.rs

```diff
@@ -745,7 +745,18 @@ macro_rules! uint_impl {
         #[stable(feature = "rust1", since = "1.0.0")]
         #[inline]
         pub fn trailing_zeros(self) -> u32 {
-            unsafe { $cttz(self as $ActualT) as u32 }
+            // As of LLVM 3.6 the codegen for the zero-safe cttz8 intrinsic emits two conditional
+            // moves on x86_64. By promoting the value to u32 and setting bit 8, we get better code
```

Comment:
This comment may be a bit out of date (it's promoted to u16 now, right?). Also, can you wrap this to 80 chars as well? (surrounding style)